packages/calcite-components/src/components/dialog/dialog.scss

```diff
@@ -139,31 +138,26 @@ calcite-panel {
   -webkit-overflow-scrolling: touch;
   visibility: hidden;
   transition:
-    transform var(--calcite-internal-animation-timing-slow) $easing-function,
-    visibility 0ms linear var(--calcite-internal-animation-timing-slow),
+    inset-block var(--calcite-internal-animation-timing-slow) $easing-function,
     opacity var(--calcite-internal-animation-timing-slow) $easing-function;
 
-  --calcite-internal-dialog-hidden-position: translate3d(0, 20px, 0);
-  --calcite-internal-dialog-shown-position: translate3d(0, 0, 0);
+  min-inline-size: var(--calcite-dialog-min-size-x, var(--calcite-internal-dialog-min-size-x));
+  min-block-size: var(--calcite-dialog-min-size-y, var(--calcite-internal-dialog-min-size-y));
 
-  inset-inline-start: var(--calcite-dialog-offset-x);
-  inset-block-start: var(--calcite-dialog-offset-y);
+  --calcite-internal-dialog-hidden-position: calc(var(--calcite-dialog-offset-y, 0px) + 20px);
+  --calcite-internal-dialog-shown-position: var(--calcite-dialog-offset-y, 0);
+
+  inset-inline-start: var(--calcite-dialog-offset-x, 0);
 
   &--opening {
     &-idle {
-      transform: var(--calcite-internal-dialog-hidden-position);
+      inset-block-start: var(--calcite-internal-dialog-hidden-position);
     }
     &-active {
-      transform: var(--calcite-internal-dialog-shown-position);
+      inset-block-start: var(--calcite-internal-dialog-shown-position);
     }
-  }
-
-  &--closing {
-    &-idle {
-      transform: var(--calcite-internal-dialog-shown-position);
-    }
-    &-active {
-      transform: var(--calcite-internal-dialog-hidden-position);
+    &-done {
```

Comment:
added a done class to handle animation when an offset is set.

Reply:
Maybe it's the naming that's throwing me off, but what's the difference between `idle` and `done`? For context, the `idle` and `active` classes match a pattern commonly used [in](https://maquettejs.org/docs/animations.html#css-transitions) [transition](https://sijakret.github.io/lit-transition/css-transitions) [libs](https://reactcommunity.org/react-transition-group/css-transition) where the `idle` class is meant to set the initial transition state and the `active` class defines the end state. I'll create a wiki page on this since I don't think this is documented anywhere. We can also revisit naming if it helps.